x-pack/plugins/maps/public/shared/layers/vector_layer.js

```diff
@@ -502,16 +502,12 @@ export class VectorLayer extends AbstractLayer {
   }
 
   async getPropertiesForTooltip(properties) {
-    const tooltipsFromSource =  await this._source.filterAndFormatProperties(properties);
-
-    //add tooltips from joins
-    return this._joins.reduce((acc, join) => {
-      const propsFromJoin = join.filterAndFormatPropertiesForTooltip(properties);
-      return {
-        ...propsFromJoin,
-        ...acc,
-      };
-    }, { ...tooltipsFromSource });
+    const tooltipsFromSource =  await this._source.filterAndFormatPropertiesToHtml(properties);
+    for (let i = 0; i < this._joins.length; i++) {
+      const propsFromJoin = await this._joins[i].filterAndFormatPropertiesForTooltip(properties);
+      Object.assign(tooltipsFromSource, propsFromJoin);
```

Comment:
this flips priority order compared to earlier. I don't think it matters, but it just means later props will override earlier props if they have the same name (I don't think this can arise either way though)